### ISSUE TITLE
Annotation GFFs missing AHRD-generated Note attributes

### ISSUE THREAD

**Issue by sammyjava** [legumeinfo/datastore-issues#168]
The following GFFs either lack Note: attributes or have none containing GO: terms. (@maxglycine it is indeed a common problem with a number of Glycine and Medicago annotations.) Please fix.

- [ ] Arabidopsis/thaliana/annotations/Col0.gnm9.ann11.KH24/arath.Col0.gnm9.ann11.KH24.gene_models_main.gff3.gz
- [x] #169
- [x] #170
- [ ] Faidherbia/albida/annotations/WAFC.gnm1.ann1.RTP9/faial.WAFC.gnm1.ann1.RTP9.gene_models_main.gff3.gz
- [x] Glycine/max/annotations/FiskebyIII.gnm1.ann1.SS25/glyma.FiskebyIII.gnm1.ann1.SS25.gene_models_main.gff3.gz
- [x] Glycine/max/annotations/Hefeng25_IGA1002.gnm1.ann1.320V/glyma.Hefeng25_IGA1002.gnm1.ann1.320V.gene_models_main.gff3.gz
- [x] Glycine/max/annotations/Huaxia3_IGA1007.gnm1.ann1.LKC7/glyma.Huaxia3_IGA1007.gnm1.ann1.LKC7.gene_models_main.gff3.gz
- [x] Glycine/max/annotations/Jinyuan_IGA1006.gnm1.ann1.2NNX/glyma.Jinyuan_IGA1006.gnm1.ann1.2NNX.gene_models_main.gff3.gz
- [x] Glycine/max/annotations/Wenfeng7_IGA1001.gnm1.ann1.ZK5W/glyma.Wenfeng7_IGA1001.gnm1.ann1.ZK5W.gene_models_main.gff3.gz
- [x] Glycine/max/annotations/Wm82_IGA1008.gnm1.ann1.FGN6/glyma.Wm82_IGA1008.gnm1.ann1.FGN6.gene_models_main.gff3.gz
- [x] Glycine/max/annotations/Wm82_ISU01.gnm2.ann1.FGFB/glyma.Wm82_ISU01.gnm2.ann1.FGFB.gene_models_main.gff3.gz
- [x] Glycine/max/annotations/Wm82.gnm1.ann1.DvBy/glyma.Wm82.gnm1.ann1.DvBy.gene_models_main.gff3.gz
- [x] Glycine/max/annotations/Zh13_IGA1005.gnm1.ann1.87Z5/glyma.Zh13_IGA1005.gnm1.ann1.87Z5.gene_models_main.gff3.gz
- [x] Glycine/max/annotations/Zh13.gnm2.ann1.FJ3G/glyma.Zh13.gnm2.ann1.FJ3G.gene_models_main.gff3.gz
- [x] Glycine/max/annotations/Zh35_IGA1004.gnm1.ann1.RGN6/glyma.Zh35_IGA1004.gnm1.ann1.RGN6.gene_models_main.gff3.gz
- [x] Glycine/soja/annotations/F_IGA1003.gnm1.ann1.G61B/glyso.F_IGA1003.gnm1.ann1.G61B.gene_models_main.gff3.gz
- [ ] Lotus/japonicus/annotations/lotja.MG20.gnm3.ann1.WF9B.gene_models_main.gff3.gz
- [x] Medicago/truncatula/annotations/HM004.gnm1.ann1.2XTB/medtr.HM004.gnm1.ann1.2XTB.gene_models_main.gff3.gz
- [x] Medicago/truncatula/annotations/HM010.gnm1.ann1.WV9J/medtr.HM010.gnm1.ann1.WV9J.gene_models_main.gff3.gz
- [x] Medicago/truncatula/annotations/HM022.gnm1.ann1.6C8N/medtr.HM022.gnm1.ann1.6C8N.gene_models_main.gff3.gz
- [x] Medicago/truncatula/annotations/HM023.gnm1.ann1.WZN8/medtr.HM023.gnm1.ann1.WZN8.gene_models_main.gff3.gz
- [x] Medicago/truncatula/annotations/HM034.gnm1.ann1.YR6S/medtr.HM034.gnm1.ann1.YR6S.gene_models_main.gff3.gz
- [x] Medicago/truncatula/annotations/HM050.gnm1.ann1.GWRX/medtr.HM050.gnm1.ann1.GWRX.gene_models_main.gff3.gz
- [x] Medicago/truncatula/annotations/HM056.gnm1.ann1.CHP6/medtr.HM056.gnm1.ann1.CHP6.gene_models_main.gff3.gz
- [x] Medicago/truncatula/annotations/HM058.gnm1.ann1.LXPZ/medtr.HM058.gnm1.ann1.LXPZ.gene_models_main.gff3.gz
- [x] Medicago/truncatula/annotations/HM060.gnm1.ann1.H41P/medtr.HM060.gnm1.ann1.H41P.gene_models_main.gff3.gz
- [x] Medicago/truncatula/annotations/HM095.gnm1.ann1.55W4/medtr.HM095.gnm1.ann1.55W4.gene_models_main.gff3.gz
- [x] Medicago/truncatula/annotations/HM125.gnm1.ann1.KY5W/medtr.HM125.gnm1.ann1.KY5W.gene_models_main.gff3.gz
- [x] Medicago/truncatula/annotations/HM129.gnm1.ann1.7FTD/medtr.HM129.gnm1.ann1.7FTD.gene_models_main.gff3.gz
- [x] Medicago/truncatula/annotations/HM185.gnm1.ann1.GB3D/medtr.HM185.gnm1.ann1.GB3D.gene_models_main.gff3.gz
- [x] Medicago/truncatula/annotations/HM324.gnm1.ann1.SQH2/medtr.HM324.gnm1.ann1.SQH2.gene_models_main.gff3.gz
- [ ] Prunus/persica/annotations/Lovell.gnm2.ann1.S2ZZ/prupe.Lovell.gnm2.ann1.S2ZZ.gene_models_main.gff3.gz
- [ ] Vitis/vinifera/annotations/PN40024.gnm2.ann1.V31M/vitvi.PN40024.gnm2.ann1.V31M.gene_models_main.gff3.gz

**Comment by adf-ncgr** [comment]
Thanks, your list looks like what I came up with independently. Do you think there's any reason to deal with the non-legumes- they don't get loaded into the mines do they? I could see some argument being made for datastore consistency, but would feel a bit barbaric trampling on the Arabidopsis thaliana annotations!

**Comment by sammyjava** [comment]
Yeah, I don't load non-legumes so no fuss on those. I just posted an RFO essentially to that effect - that the presence 
of GO terms will be validated at mine loading.

**Comment by adf-ncgr** [comment]
I believe all of the Glycine genomes are now ready to go. I will deal with Medicago next but whenever you're ready to update GlycineMine hopefully the datastore will comply (especially whatever genome Rex was looking at that started this whole thing!)

**Comment by sammyjava** [comment]
@adf-ncgr I just added Lotus/japonicus/annotations/lotja.MG20.gnm3.ann1.WF9B.gene_models_main.gff3.gz to the list. Came up when loading LegumeMine.

[convertFile] ## Validating lotja collection MG20.gnm3.ann1.WF9B
[convertFile]  - lotja.MG20.gnm3.ann1.WF9B.gene_models_main.gff3.gz
[convertFile] ## INVALID: lotja.MG20.gnm3.ann1.WF9B.gene_models_main.gff3.gz gene records do not contain the Note attribute.
[convertFile] ## INVALID: lotja.MG20.gnm3.ann1.WF9B.gene_models_main.gff3.gz gene record Note attributes do not contain GO terms.

**Comment by sammyjava** [comment]
I'm about to start post-processing LegumeMine without MG20.gnm3.ann1.WF9B, which is a long post-process. Hold me off if you think you'll have that annotation set fixed soon-ish. The only other remaining mine to be built on 5.1.0.3 is GlycineMine, with a couple straggler sources to be loaded.

**Comment by adf-ncgr** [comment]
Yeah, why don't you hold off - without making any guarantees about this specific annotation I'd like to make sure I'm on the same page about what all has been included so far- what's the easiest way for me to look at that?

**Comment by sammyjava** [comment]
Use this alias on lupini-mines to see all loaded datasets in a mine db:
```
alias datasets='psql -c "SELECT name FROM dataset ORDER BY name"'
```
usage: `datasets legumemine-5103`

Use this handy script IN THE MINE DIR to see what's in project.xml but NOT loaded:
```
/home/shokin/bin/datasets.sh <db-name>
```
Caveat: project.xml source names can't have extended ASCII so, for example, vignamine will show a bunch that are actually loaded because the dataset name has high ASCII but the source name does not.
```
[shokin@lupini-mines vignamine]$ missing-datasets.sh vignamine-5103
-----------------------------------------------------------------
NOTE: datasets with high ASCII characters will appear as missing!
-----------------------------------------------------------------
mixed.gwas.Lo_Munoz-Amatriain_2019
UCR-minicore.gwas.Herniter_Munoz-Amatriain_2018
CB27_x_IT82E-18.map.Herniter_Munoz-Amatriain_2018
iSelect-consensus-2016.map.Munoz-Amatriain_Mirebrahim_2017
IT99K-573-1-1_x_TVNu-1158.map.Lo_Munoz-Amatriain_2018
Sanzi_x_Vita7.map.Herniter_Munoz-Amatriain_2018
IT99K-573-1-1_x_TVNu-1158.qtl.Lo_Munoz-Amatriain_2018
mixed.qtl.Herniter_Munoz-Amatriain_2018
```
ALSO: I've switched to keeping the suffix on production databases (I no longer rename them). So the new mines are running on the `mine-5103` databases in production. Same deal with the github branches, `main` is no longer relevant.

There is a legumista account which you can use, and create your own if you like. Take a look at my .pgpass (sudo required) to see how I connect to papadum. Nothing new. I also have
```
export PGHOST=papadum
export PGUSER=intermine
```
in .bashrc to keep things simple.

**Comment by adf-ncgr** [comment]
OK @sammyjava per our discussion I think lotja.MG20.gnm3.ann1.WF9B.gene_models_main.gff3.gz is ready for another try. Note that per our discussion I demoted some genes into a separate low quality gff file (namely those not contained in the protein file). Hopefully I didn't forget some file in which those things may pop up. Let me know if you run into further issues

**Comment by sammyjava** [comment]
New files fail validation.
```
## Validating lotja collection MG20.gnm3.ann1.WF9B
 - lotja.MG20.gnm3.ann1.WF9B.gene_models_main.gff3.gz
10:00:28 [main] INFO  org.biojava.nbio.genome.parsers.gff.GFF3Reader - Reading: /tmp/temp.gff3
## INVALID: lotja.MG20.gnm3.ann1.WF9B.gene_models_main.gff3.gz record ID is duplicate of one already read:
## INVALID: lotja.MG20.gnm3.Lj0	protein_coding	exon	-17295290	-17294247	100.0	.	ID=lotja.MG20.gnm3.ann1.Lj0g3v0050909.1.exon.1;Parent=lotja.MG20.gnm3.ann1.Lj0g3v0050909.1;Type="protein_coding"
## INVALID: lotja.MG20.gnm3.ann1.WF9B.gene_models_main.gff3.gz record ID attribute is missing or invalid:
## INVALID: lotja.MG20.gnm3.Lj0	protein_coding	exon	159887798	159887899	100.0	.	ID=known_Npl.mrna1.exon1;Name=known_Npl:3;Parent=lotja.MG20.gnm3.ann1.Lj0g3v0306419.1;Target=known_Npl 1 101 +
## INVALID: {Target=known_Npl 1 101 +, Parent=lotja.MG20.gnm3.ann1.Lj0g3v0306419.1, ID=known_Npl.mrna1.exon1, Name=known_Npl:3}
 x lotja.MG20.gnm3.ann1.WF9B.gene_models_main.gff3.gz 20227 gene record Note attributes are missing GO terms.
 - lotja.MG20.gnm3.ann1.WF9B.protein.faa.gz
 - lotja.MG20.gnm3.ann1.WF9B.protein_primary.faa.gz
 - lotja.MG20.gnm3.ann1.WF9B.cds.fna.gz
 - lotja.MG20.gnm3.ann1.WF9B.cds_primary.fna.gz
 - lotja.MG20.gnm3.ann1.WF9B.mrna.fna.gz
## INVALID: lotja.MG20.gnm3.ann1.WF9B.mrna.fna.gz has an invalid sequence identifier in header:
AB433810.1
 - lotja.MG20.gnm3.ann1.WF9B.mrna_primary.fna.gz
 - lotja.MG20.gnm3.ann1.WF9B.iprscan.gff3.gz
10:00:46 [main] INFO  org.biojava.nbio.genome.parsers.gff.GFF3Reader - Reading: /tmp/temp.gff3
## INVALID: lotja.MG20.gnm3.ann1.WF9B.iprscan.gff3.gz has an invalid seqname:
## INVALID: AB433810.1	PANTHER	protein_match	80	261	0.0	.	Name=PTHR19384;status=T;ID=match$3042_81_261;date=09-08-2023
 - lotja.MG20.gnm3.ann1.WF9B.legfed_v1_0.M65K.gfa.tsv.gz
 - lotja.MG20.gnm3.ann1.WF9B.phytozome_10_2.HFNR.gfa.tsv.gz
```

**Comment by adf-ncgr** [comment]
OK, these annotations are more problematic than I'd imagined. Better move ahead on legumemine without this dataset.
